src/AclExtras.php

```diff
@@ -303,7 +303,7 @@ protected function _processPlugins($root, array $plugins = [])
      * @param string $prefix Name of the prefix you are making controllers for.
      * @return array
      */
-    protected function _updateControllers($root, $controllers, $plugin = null, $prefix = null)
+    protected function _updateControllers($root, array $controllers, string $plugin = '', string $prefix = '')
```

Comment:
Why not keep `$plugin` and `$prefix` nullable? These were nullable in the past and passing an empty string seems wrong here.

Reply:
They get passed to `Inflector::camelize()` which requires to get a string now. I assume in Cake 3 `Inflector::camelize(null)` will return an empty string `''`.

So why passing `null` as default if the application makes an empty string out of it anyway.

Reply:
You can just check for null before inflecting probably.
Might be a bit cleaner then.